pkg/component/shared/istio.go

```diff
@@ -239,6 +259,37 @@ func IsZonalIstioExtension(labels map[string]string) (bool, string) {
 	return false, ""
 }
 
+// ShouldEnsureHostSpreading checks whether all given zones have at least two nodes so that Istio can be spread across hosts in each zone
+func ShouldEnsureHostSpreading(ctx context.Context, cl client.Client, zones []string) (bool, error) {
```

Comment:
Can you add a unit test for this function, please? Especially the `for` loop looks dangerous. Maybe it makes even sense to split this up a bit into smaller helper functions to reduce the complexity there.

Reply:
Added test in https://github.com/gardener/gardener/pull/8876/commits/b46717d4148cbf3a08e82cb057f78074e60665ea.

Do you think that this is sufficient? The loops are only to count the number of nodes per zone including a short-cut for when we already have enough information per zone. I am open to reworking it so that it looks less dangerous.